Yosemite/YosemiteTests/Stores/AppSettingsStoreTests.swift

```diff
@@ -0,0 +1,143 @@
+import XCTest
+@testable import Yosemite
+@testable import Storage
+
+
+/// Mock constants
+///
+private struct TestConstants {
+    static let fileURL = Bundle(for: AppSettingsStoreTests.self)
+        .url(forResource: "shipment-provider", withExtension: "plist")
+    static let siteID = 156590080
+    static let providerName = "post.at"
+
+    static let newSiteID = 1234
+    static let newProviderName = "Some provider"
+}
+
+
+/// AppSettingsStore unit tests
+///
+final class AppSettingsStoreTests: XCTestCase {
+    /// Mockup Dispatcher!
+    ///
+    private var dispatcher: Dispatcher?
+
+    /// Mockup Storage: InMemory
+    ///
+    private var storageManager: MockupStorageManager?
+
+    /// Mockup File Storage: Load a plist in the test bundle
+    ///
+    private var fileStorage: MockFileLoader?
+
+    /// Test subject
+    ///
+    private var subject: AppSettingsStore?
+
+    override func setUp() {
+        super.setUp()
+        dispatcher = Dispatcher()
+        storageManager = MockupStorageManager()
+        fileStorage = MockFileLoader()
+        subject = AppSettingsStore(dispatcher: dispatcher!, storageManager: storageManager!, fileStorage: fileStorage!)
+        subject?.selectedProvidersURL = TestConstants.fileURL!
+    }
+
+    override func tearDown() {
+        dispatcher = nil
+        storageManager = nil
+        fileStorage = nil
+        subject = nil
+        super.tearDown()
+    }
+
```

Comment:
Can you also test that failures throw the expected error, such as `writePreselectedProvider`?

Reply:
Thanks for catching that